api/wasm/cpp/proxy_wasm_intrinsics.h

```diff
@@ -999,3 +1188,194 @@ inline Histogram<Tags...>* Histogram<Tags...>::New(std::string_view name,
   return new Histogram<Tags...>(std::string(name),
                                 std::vector<MetricTag>({ToMetricTag(descriptors)...}));
 }
+
+inline uint32_t grpcCall(std::string_view service, std::string_view service_name, std::string_view method_name,
+    const google::protobuf::MessageLite &request, uint32_t timeout_milliseconds) {
+  std::string serialized_request;
+  request.SerializeToString(&serialized_request);
+  return proxy_grpcCall(service.data(), service.size(), service_name.data(), service_name.size(), method_name.data(), method_name.size(),
+      serialized_request.data(), serialized_request.size(), timeout_milliseconds);
+                         
+}
+
+inline uint32_t grpcStream(std::string_view service, std::string_view service_name, std::string_view method_name) {
+  return proxy_grpcStream(service.data(), service.size(), service_name.data(), service_name.size(), method_name.data(), method_name.size());
+}
+
+inline void grpcCancel(uint32_t token) {
+  return proxy_grpcCancel(token);
+}
+
+inline void grpcClose(uint32_t token) {
+  return proxy_grpcClose(token);
+}
+
+inline void grpcSend(uint32_t token, std::string_view message, bool end_stream) {
+  return proxy_grpcSend(token, message.data(), message.size(), end_stream ? 1 : 0);
+}
+
+inline void Context::httpCall(std::string_view uri, const HeaderStringPairs& request_headers,
+    std::string_view request_body, const HeaderStringPairs& request_trailers,
+    uint32_t timeout_milliseconds, HttpCallCallback callback) {
+  auto token = ::httpCall(uri, request_headers, request_body, request_trailers, timeout_milliseconds);
+  if (token) {
+    http_calls_[token] = std::move(callback);
+  } else {
+    throw ProxyException("httpCall failed");
+  }
+}
+
+inline void Context::onHttpCallResponse(uint32_t token, std::unique_ptr<WasmData> header_pairs,
+    std::unique_ptr<WasmData> body, std::unique_ptr<WasmData> trailer_pairs) {
+  auto it = http_calls_.find(token);
+  if (it != http_calls_.end()) {
+    it->second(std::move(header_pairs), std::move(body), std::move(trailer_pairs));
+    http_calls_.erase(token);
+  }
+}
+
+inline void Context::grpcSimpleCall(std::string_view service, std::string_view service_name, std::string_view method_name,
+    const google::protobuf::MessageLite &request, uint32_t timeout_milliseconds, Context::GrpcSimpleCallCallback callback) {
+  auto token = grpcCall(service, service_name, method_name, request, timeout_milliseconds);
+  if (token) {
+    simple_grpc_calls_[token] = std::move(callback);
```

Comment:
This statement causes a crash, though it seems perfectly right for me:

```
Caught exception "wavm.integerDivideByZeroOrOverflow
Call stack:
  host!envoy!<unknown>
  wasm!!__ZNSt3__213unordered_mapIjNS_8functionIFv10GrpcStatusNS_10unique_ptrI8WasmDataNS_14default_deleteIS4_EEEEEEENS_4hashIjEENS_8equal_toIjEENS_9allocatorINS_4pairIKjS9_EEEEEixERSG_+0
  wasm!!__ZN7Context14grpcSimpleCallENSt3__217basic_string_viewIcNS0_11char_traitsIcEEEES4_S4_RKN6google8protobuf11MessageLiteEjNS0_8functionIFv10GrpcStatusNS0_10unique_ptrI8WasmDataNS0_14default_deleteISD_EEEEEEE+138
  wasm!!__ZN7Context14grpcSimpleCallIN6google8protobuf5EmptyEEEvNSt3__217basic_string_viewIcNS4_11char_traitsIcEEEES8_S8_RKNS2_11MessageLiteEjNS4_8functionIFvOT_EEENSC_IFv10GrpcStatusS8_EEE+335
....
....
```

I tried to dig a bit more into it, and experiment with making a simpler function map `unordered_map<uint32_t, std::function<void()>>` and insert a void() function into it, crash still happens. I work around it by commenting this line out, but the issue is still not clear to me..